backend/pubsub.js

```diff
@@ -301,7 +305,31 @@ const defaultMessageHandlers = {
     } else {
       log.debug('Already subscribed to', channelID)
     }
-    socket.send(createOkResponse({ type: SUB, channelID }))
+    // $FlowFixMe[incompatible-call]
+    socket.send(createOkResponse({ type: SUB, channelID, kvFilter }))
+  },
+
+  [KV_FILTER] ({ channelID, kvFilter }: SubMessage) {
+    const socket = this
+    const { server } = this
+
+    if (!server.channels.has(channelID)) {
+      socket.send(createErrorResponse(
+        { type: SUB, channelID, reason: `Unknown channel id: ${channelID}` }
+      ))
+      return
+    }
+    if (socket.subscriptions.has(channelID)) {
+      if (Array.isArray(kvFilter)) {
+        socket.kvFilter.set(channelID, new Set(kvFilter))
+      } else {
+        socket.kvFilter.delete(channelID)
+      }
+    } else {
+      log.debug('Not subscribed to', channelID)
```

Comment:
Could you prefix this string with something like `[KV_FILTER]` to give it more context?